src/components/IconToggleButton.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { StyleSheet, View } from 'react-native';
+
+import { palette } from '../styles';
+import { Icon } from './Icon';
+import { IconButton } from './IconButton';
+
+interface Props {
+  icons: string[]; // icons name
+  onPress: (value: boolean) => void;
+}
+
+interface State {
+  isFirstOn: boolean;
+}
+
+export class IconToggleButton extends React.PureComponent<Props, State> {
+  state = {
+    isFirstOn: true,
+  };
+
+  handlePressFirst = () => {
+    this.setState({
+      isFirstOn: true,
+    });
+    this.props.onPress(true);
+  };
+
+  handlePressSecond = () => {
+    this.setState({
+      isFirstOn: false,
+    });
+    this.props.onPress(false);
+  };
+
+  render() {
+    const [firstIcon, secondIcon] = this.props.icons;
+
+    return (
+      <View style={styles.toggleButtonContainer}>
+        <View>
+          <IconButton
+            name={firstIcon}
+            type="material"
+            onPress={this.handlePressFirst}
+            size={24}
+            color={palette.primaryVariant}
+          />
+          <View style={this.state.isFirstOn ? styles.buttonOn : styles.buttonOff}>
+            <Icon name={firstIcon} type="material" size={24} color={palette.primaryVariant} />
+          </View>
+        </View>
+        <View>
+          <IconButton
+            name={secondIcon}
+            type="material"
+            onPress={this.handlePressSecond}
+            size={24}
+            color={palette.primaryVariant}
+          />
+          <View style={!this.state.isFirstOn ? styles.buttonOn : styles.buttonOff}>
```

Comment:
Try adjust `IconButton` style to get rid conditions. Maybe you shoud add `zIndex` there.